.github/workflows/changelog.yml

```diff
@@ -12,3 +12,7 @@ jobs:
           changelog_regex: "CHANGELOG.md"
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Changelog Tool
+        uses: rcmachado/changelog-action@v1
+        with:
+          args: diff -u <(changelog fmt -f CHANGELOG.md) CHANGELOG.md
```

Comment:
@Pradhvan the problem here is because `args` is what is being passed as the argument to the `changelog` command. This `diff` you want is system's. The way it is implemented here, GitHub Actions is going to call:

```
$ changelog diff -u <(changelog fmt -f CHANGELOG.md) CHANGELOG.md 
```

... resulting in an error given that the tool has no argument called `diff`.

There are a few ways to solve this:
1. Implement the `diff` command on @rcmachado 's tool (or as he points out in https://github.com/rcmachado/changelog/issues/71: `fmt --check`)
2. Do not use his GitHub Action and simply download and execute the command yourself
3. Keep using his GitHub Action but try to play with shell arguments to execute `fmt`, get the results from it, and then call system's diff

IMO:
- 1 is the bestion option if you know (or want to play with) Golang.
- 3 might work well if you know how to use `xargs` (I don't 😅)
- 2 could be achieved by some sort of the following commands:

```yml
- name: Changelog Tool
- run: |
  go get github.com/rcmachado/changelog
  diff -u <(./bin/changelog fmt -f CHANGELOG.md) CHANGELOG.md
```

Reply:
@barbosa fun thing is I actually picked up Go and will go for 1 😄